### ISSUE TITLE
The size of the generated model varies

### ISSUE THREAD

**Issue by Mrxiaofei** [NVIDIA/TensorRT#1071]
## Description

Caffemodel ResNet50 transformation using trtexec，after many transformations, it is found that the model size is changing.The different size of the model generated, the forward inference memory is different,but accuracy is same.Also I had tried pth model, the same question exists!

## Environment

**TensorRT Version**: 7.2.2 and 7.0(both used)
**NVIDIA GPU**: 1080
**NVIDIA Driver Version**: 418.56( and 440.33.01)
**CUDA Version**: 10.1 for tensorRT7.0 and 10.2 for tensorRT7.2
**CUDNN Version**: 7.5.1 for tensorRT7.0 and cudnn 8 for tensorRT7.2
**Operating System**: Linux
**Python Version (if applicable)**: 
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: pytorch 1.4
**Baremetal or Container (if so, version)**: 

## Relevant Files

Caffemodel ResNet50 is used by tensorRT tar package included.
pytorch model is used by https://github.com/Mrxiaofei/tensorrtx/tree/master/resnet

## Steps To Reproduce

Just repeatly convert the model and record the model size,you will find the question.
Looking forward to your reply

**Comment by ttyio** [comment]
Hello @Mrxiaofei , thanks for reporting.
The model size depends on the tactics selected when you build the tensorrt engine. Different tactic has impact on memory consumption and end to end runtime latency. So we recommend user to build engine in the same environment as the final deploy environment. And the non-deterministic might come from the floating gpu clock, you might find the result is more stable if you lock your gpu clock.

**Comment by Mrxiaofei** [comment]
Hello@ttyio，thank you for your reply.
I konw why the size of model changes.There's something else that confuses me.First, will different combinations of tactic affect the accuracy(fp32 data type)? Or the combination just affect memory consumption and end to end runtime latency.Also you mean the result is more stable if you lock your gpu clock.So will the results vary with the performance of the current machine?If I run some other programs when I convert the model,the result will be worse?

**Comment by ttyio** [comment]
Hello @Mrxiaofei 
> will different combinations of tactic affect the accuracy(fp32 data type)
It might, because floating point has rounding issue when arithmetic operation sequence changed. 

> you mean the result is more stable if you lock your gpu clock.So will the results vary with the performance of the current machine.
Here the stable means that you always get the same tactic when you build engine multiple times. However, deep learning is by nature somewhat robust again the small rounding errors. that's why FP16/INT8 inference also works well. So mostly the network accuracy is not impacted when selecting different tactic, we only see end to end performance changes. If we see some tactic cause accuracy issue, we can use AlgorithmSelector to disable specific tactic. Check usage of AlgorithmSelector in https://github.com/NVIDIA/TensorRT/tree/master/samples/opensource/sampleAlgorithmSelector

**Comment by Mrxiaofei** [comment]
Hello @ttyio ,thanks for your explanation!
